### DISCUSSION TITLE
Interactive images

### DISCUSSION THREAD

**Discussion by bantin** [marimo-team/marimo#319] [General]
I do a lot of work processing videos for neuroscience data. Often, we will want to look at a summary image (say, the mean over time of a long video), and then click on an individual pixel in that image to bring up the time series for that specific pixel. This is something that is very difficult in jupyter. 

I'm curious if Marimo supports a feature like this?

**Comment by akshayka** [comment]
Hi @bantin! Thanks for the question. We can definitely support that in the future. That *may* be possible today with mo.ui.altair_chart, or with our upcoming reactive plotly element. I will investigate.

To help me better understand the UI you are imagining:

How many pixels are typically in these images? For a high resolution image I imagine it's difficult to select any particular pixel with precision.

**Comment by mscolnick** [comment]
@bantin, i tried this out with [plotly](https://plotly.com/python/) and our new `mo.ui.ploty` wrapper (which binds the frontend values to the Python values), and it works pretty well if you want to give it a shot.

![Screenshot 2023-11-28 at 5 42 14 PM](https://github.com/marimo-team/marimo/assets/2753772/bf00659f-7c36-45be-8334-fab9bbad164e)

[Here is the example](https://github.com/marimo-team/marimo/blob/main/examples/third_party/plotly_image_selection.py)

To run the example code, run:
```bash
marimo edit https://github.com/marimo-team/marimo/blob/main/examples/third_party/plotly_image_selection.py
```

Let us know if this captures your use-case or if there are ways we can further make this possible.
